streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -480,6 +482,11 @@ public Set<String> makeReady(final Map<String, InternalTopicConfig> topics) {
                     .filter(e -> topicsNotReady.contains(e.getKey()))
                     .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
             final Set<NewTopic> topicsToCreate = computeTopicsToCreate(notReadyTopicsMap, tempUnknownTopics);
+
+            topicsNotReady.retainAll(tempUnknownTopics);
```

Comment:
Why do we remove unknown topics? If a topic is unknown it is by definition not ready, and we should keep it in `topicsNotReady`.